message.go

```diff
@@ -500,13 +503,11 @@ func (m *Message) Unmarshal(v interface{}) error {
 			if dataField.IsNil() && dataField.Kind() != reflect.Slice {
 				dataField.Set(reflect.New(dataField.Type().Elem()))
 			}
-			err := messageField.Unmarshal(dataField.Interface())
-			if err != nil {
+			if err := messageField.Unmarshal(dataField.Interface()); err != nil {
```

Comment:
meganit: generally I would avoid driveby changes in an open source repo so that the change history is clearer as each PR has a single focus